pkg/nsm/apis/nsmconnect/nsmconnect.proto

```diff
@@ -0,0 +1,57 @@
+syntax = "proto3";
+package nsmconnect;
+
+message Label {
+    map<string ,string> selector = 1;
+}
+
+message Metadata {
+    string Name = 1;
+    string Namespace = 2;
+    Label Labels = 3;
+}
+
+enum InterfaceType {
+    KernelInterface = 0;
+    VhostUser = 1;
+    Memif = 2;
+    Other = 3;
+  }
+
+message MechanismType {
+    InterfaceType type = 1;
+  }
+
+message AffinityPreference {
+    string Predicate = 1;
+}
+
+message ConnectionRequest {
+
+    Metadata Metadata = 1;
+    
+    string NetworkServiceName = 2;
+
+    repeated MechanismType Type = 3;
+
+    repeated AffinityPreference Predicate = 4;
+
+    string DeviceClaim = 5;
+
+}
+
+message ConnectionParameters {
+    string Addressing = 1;
+    repeated string Route = 2;
+}
+
+message ConnectionAccept {
+    bool Accepted = 1;
+    string Error = 2;
+    MechanismType Type = 3;
```

Comment:
We will also need Mechanism parameters here.

If I'm asking for a Kernel Interface, I need to know the interface name within my Pod's namespace
If I'm asking for a memif or vhostuser I need to know the specific socket name (they get set up over unix file sockets) etc. Mechanism parameters will vary by MechanismType.

Reply:
each mechanism type can be represented by a struct with relevant to this type parameters, would it work for you?